packages/netlify-cms-backend-bitbucket/src/implementation.ts

```diff
@@ -255,27 +254,67 @@ export default class BitbucketBackend implements Implementation {
     const listFiles = () =>
       this.api!.listFiles(folder, depth).then(({ entries, cursor: c }) => {
         cursor = c.mergeMeta({ extension });
-        return filterByPropExtension(extension, 'path')(entries);
+        return entries.filter(e => filterByExtension(e, extension));
       });
 
-    const files = await entriesByFolder(listFiles, this.api!.readFile.bind(this.api!), 'BitBucket');
+    const head = await this.api!.defaultBranchCommitSha();
+    const readFile = (path: string, id: string | null | undefined) => {
+      return this.api!.readFile(path, id, { head }) as Promise<string>;
+    };
+
+    const files = await entriesByFolder(
+      listFiles,
+      readFile,
+      this.api!.readFileMetadata.bind(this.api),
+      API_NAME,
+    );
 
     // eslint-disable-next-line @typescript-eslint/ban-ts-ignore
     // @ts-ignore
     files[CURSOR_COMPATIBILITY_SYMBOL] = cursor;
     return files;
   }
 
+  async listAllFiles(folder: string, extension: string, depth: number) {
+    const files = await this.api!.listAllFiles(folder, depth);
+    const filtered = files.filter(file => filterByExtension(file, extension));
+    return filtered;
+  }
+
   async allEntriesByFolder(folder: string, extension: string, depth: number) {
-    const listFiles = () =>
-      this.api!.listAllFiles(folder, depth).then(filterByPropExtension(extension, 'path'));
+    const head = await this.api!.defaultBranchCommitSha();
+
+    const readFile = (path: string, id: string | null | undefined) => {
+      return this.api!.readFile(path, id, { head }) as Promise<string>;
+    };
 
-    const files = await entriesByFolder(listFiles, this.api!.readFile.bind(this.api!), 'BitBucket');
+    const files = await allEntriesByFolder({
```

Comment:
Yes this is ugly, but at least it is shared between GitLab and Bitbucket.
We can do some cleanup later and consolidate all the file listing methods into one